api/src/main/java/org/eclipse/microprofile/health/spi/SPIFactory.java

```diff
@@ -19,14 +19,16 @@
  * SPDX-License-Identifier: Apache-2.0
  *
  */
-package org.eclipse.microprofile.health;
+
+package org.eclipse.microprofile.health.spi;
+
+import org.eclipse.microprofile.health.ResponseBuilder;
 
 /**
- * The basic health check procedure interface
- *
- * @author Heiko Braun
- * @since 13.06.17
+ * Created by hbraun on 07.07.17.
  */
-public interface HealthCheckProcedure {
-    HealthStatus perform();
+public interface SPIFactory {
+
+    ResponseBuilder createResponseBuilder();
+
```

Comment:
That's your point of view @Emily-Jiang. However @jmesnil and I think differently and hence I've decided to merge this PR. Feel free to submit proposals based on the the new API and we  take it from there.